Route2D.gd

```diff
@@ -1,9 +1,6 @@
 extends Line2D
 
-func add_new_point(position: Vector3, index: int = -1):
-	self.add_point(Vector2(position.x, position.z), index)
-	
-func update_poistion(position: Vector3, index: int):
+func on_update_position(position: Vector3, index: int):
```

Comment:
I dont think this is the best name, also why is this taking in a vector3 instead of a vector2?